src/wp-includes/rest-api/endpoints/class-wp-rest-menu-items-controller.php

```diff
@@ -101,6 +101,21 @@ protected function check_has_read_only_access( $request ) {
 		);
 	}
 
+	/**
+	 * Retrieves a collection of menu items.
+	 *
+	 * @since 6.x.x
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		add_filter( "rest_{$this->post_type}_query_results", '_prime_menu_item_objects' );
+
+		return parent::get_items( $request );
+	}
```

Comment:
I'm not sure if this is needed or not, depends how the hooks are stored when they are added. 

I'm trying to avoid adding the filter multiple times if this is called multiple times in a single request. While that's not a problem for HTTP requests to the `wp-json` endpoint, it can get called multiple times during batch requests or other techniques that use `rest_do_request()`.

```suggestion
	public function get_items( $request ) {
		add_filter( "rest_{$this->post_type}_query_results", '_prime_menu_item_objects' );
		$items = parent::get_items( $request );
		remove_filter( "rest_{$this->post_type}_query_results", '_prime_menu_item_objects' );
		
		return $items;
	}
```

Reply:
Since this is the same controller instance, the `spl_object_hash` should be the same, so it shouldn't get registered multiple times, even if the route is requested multiple times.